docs/how-to-guides/gitops-using-azure-arc-onboarding-component.md

```diff
@@ -0,0 +1,238 @@
+resource groups and tags.
+
+This guide provides the steps for onboarding a Lokomotive cluster on Azure Arc.
+
+## Learning objectives
+
+At the end of this tutorial you will have onboarded a Lokomotive cluster on Azure Arc, create an Azure Arc
+Kubernetes configuration and deploy configuration of a sample Nginx Deployment using GitOps.
+
+## Prerequisites
+
+* A Lokomotive cluster.
+
+* Azure command-line interface `az` installed on the local machine.
+
+* `jq` installed on the system.
+
+* Register providers for Azure Arc enabled Kubernetes:
+   ```bash
+   az provider register --namespace Microsoft.Kubernetes
+   az provider register --namespace Microsoft.KubernetesConfiguration
+   az provider register --namespace Microsoft.ExtendedLocation
+   ```
+
+   Monitor the registration process. Registration may take up to 10 minutes.
+
+   ```bash
+   az provider show -n Microsoft.Kubernetes -o table
+   az provider show -n Microsoft.KubernetesConfiguration -o table
+   az provider show -n Microsoft.ExtendedLocation -o table
+   ```
+
+   **NOTE**: This registration is needed only once per tenant.
+
+* Install `k8s-configuration` extension for Azure cli.
+
+  ```bash
+  az extension add --name k8s-configuration
+  ```
+
+* Create a resource group:
+
+   ```bash
+   RG_NAME="AzureArcTest"
+   az group create --name "${RG_NAME}" --location EastUS --output table
+   ```
+
+* Create a service principal, its credentials and assign roles to the service principal.
+
+  ```bash
+  # Create a ServicePrincipal and its credentials.
+  SP_NAME=azure-arc-onboarding-service-principal
+  az ad sp create-for-rbac -n "${SP_NAME}" --skip-assignment -o jsonc > /tmp/sp-cred.json
+  SP_ID=$(az ad sp list -o tsv --query='[0].objectId' --display-name "${SP_NAME}")
+
+  # Get Subscription ID
+  SUB_ID=$(az account show --query id --output tsv)
+
+  # Assign "Kubernetes Cluster - Azure Arc Onboarding" Role by its identifier
+  az role assignment create --assignee "${SP_ID}" \
+    --role "Kubernetes Cluster - Azure Arc Onboarding" \
+    --scope /subscriptions/${SUB_ID}/resourcegroups/${RG_NAME}
+
+  # Assign "Microsoft.Kubernetes connected cluster" Role by its identifier
+  az role assignment create --assignee "${SP_ID}" \
+    --role "Microsoft.Kubernetes connected cluster role" \
+    --scope /subscriptions/${SUB_ID}/resourcegroups/${RG_NAME}
+  ```
+
+## Steps
+
+### Step 1: Configure [azure-arc-onboarding](../configuration-reference/components/azure-arc-onboarding.md)
+Lokomotive component.
+
+#### Config
+
+Create a file named `azure-arc-onboarding.lokocfg` with the following contents:
+
+```bash
+
+# Copy values from /tmp/sp-cred.json created in the prerequisites section
+APPLICATION_ID=$(jq -r .appId /tmp/sp-cred.json)
+APPLICATION_PASSWORD=$(jq -r .password /tmp/sp-cred.json)
+TENANT_ID=$(jq -r .tenant /tmp/sp-cred.json)
+CLUSTER_NAME=mercury
+
+echo "component \"azure-arc-onboarding\" {
```

Comment:
I'd use `cat <<EOF > azure-arc-onboarding.lokocfg`, so you don't have to play escape game :)